diesel/src/query_dsl/mod.rs

```diff
@@ -49,6 +49,8 @@ pub use self::load_dsl::LoadQuery;
 pub use self::group_by_dsl::GroupByDsl;
 pub use self::join_dsl::{InternalJoinDsl, JoinOnDsl, JoinWithImplicitOnClause};
 pub use self::save_changes_dsl::SaveChangesDsl;
+#[doc(hidden)]
+pub use self::save_changes_dsl::InternalSaveChangesDsl;
```

Comment:
Why do we need this?

Reply:
Because otherwise it is not possible to implement it outside of diesel. (Because otherwise it is private)